From: =?utf-8?q?Tim_R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Tue, 13 Jun 2017 22:24:18 +0200
Subject: Fix memleak in _psl_is_public_suffix() (found by fuzzing)

---
 src/psl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/psl.c b/src/psl.c
index 16c7e27..ac7fcbb 100644
--- a/src/psl.c
+++ b/src/psl.c
@@ -879,7 +879,7 @@ static int _psl_is_public_suffix(const psl_ctx_t *psl, const char *domain, int t
 		_psl_entry_t *rule = _vector_get(psl->suffixes, 0);
 
 		if (!rule || rule->nlabels < suffix.nlabels - 1)
-			return 0;
+			goto suffix_no;
 
 		rule = _vector_get(psl->suffixes, _vector_find(psl->suffixes, &suffix));
 
